internal/olm/operator/internal/registry_pod.go

```diff
@@ -0,0 +1,145 @@
+package olm
+
+import (
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+	"path/filepath"
+	"time"
+
+	core "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+// RegistryPod holds resources necessary for creation of a registry server
+type RegistryPod struct {
+	registryImage       string
+	registryPodGRPCPort int32
+	registryDBPath      string
+	registryBundleImage string
+
+	ctx        context.Context
+	kubeclient kubernetes.Interface
+}
+
+// GetKubeClient creates a kubernetes clientset using kubeconfig
+func GetKubeClient() (client kubernetes.Interface) {
```

Comment:
You can assume a kubeclient will be provided to `RegistryPod` by a manager like it is for [`operatorManager`](https://github.com/operator-framework/operator-sdk/blob/master/internal/olm/operator/operator_manager.go#L144-L159).